xds/internal/xdsclient/e2e_test/federation_watchers_test.go

```diff
@@ -79,28 +80,27 @@ func setupForFederationWatchersTest(t *testing.T) (*e2e.ManagementServer, string
 	if err != nil {
 		t.Fatalf("Failed to create xDS client: %v", err)
 	}
-	return serverTestAuthority, nodeID, client
+	return serverNonDefaultAuthority, nodeID, client
 }
 
 // TestFederation_ListenerResourceContextParamOrder covers the case of watching
 // a Listener resource with the new style resource name and context parameters.
-// The test verifies the following:
-// - Two watches with the same query string, but in different order results in
-//   the two watches watching the same resource.
-// - Response with the same query string, but in different order results in both
-//   watches being notified.
+// The test registers watches for two resources which differ only in the order
+// of context parameters in their URI. The server is configured to respond with
+// a single resource with canonicalized context parameters. The test verifies
+// that both watchers are notified.
```

Comment:
So sorry, but what does "canonicalized context parameters mean"? Here and below.

Reply:
https://github.com/cncf/xds/blob/main/proposals/TP1-xds-transport-next.md#normalization is the spec. What canonicalized (or normalized) It isn't clear from reading that. But basically what it means for us is that we sort the context parameters. That is the reason why the following two resource names are considered the same:

`xdstp:///envoy.config.listener.v3.Listener/xdsclient-test-lds-resource?a=1&b=2`
`xdstp:///envoy.config.listener.v3.Listener/xdsclient-test-lds-resource?b=2&a=1`

Our federation spec does go into more detail here:
```
New-style resource names used as cache keys must be [normalized]
(https://github.com/cncf/xds/blob/main/proposals/TP1-xds-transport-next.md#normalization). In particular, this 
means sorting the context params in normal lexicographic order.

If a new-style name has duplicate values for the same context param, the behavior is undefined. The easiest thing
is probably to just use the last value seen, since this allows the implementation to just populate the map by iterating
over the query params.
```
https://github.com/grpc/proposal/blob/master/A47-xds-federation.md